src/openrct2/management/Research.cpp

```diff
@@ -910,3 +890,107 @@ bool ResearchItem::Exists() const
     }
     return false;
 }
+
+static std::bitset<RIDE_TYPE_COUNT> _seenRideType = {};
+static std::bitset<RIDE_TYPE_COUNT* MAX_RIDE_GROUPS_PER_RIDE_TYPE> _seenRideGroup = {};
+
+static void research_update_first_of_type(ResearchItem* researchItem)
```

Comment:
Is there a thing such as a `Research` class? Just trying to understand the reason for the free function in here,

Reply:
There is no Research class, no. (Most files are still a collection of C-style functions.)